src/main/scala/devices/debug/Debug.scala

```diff
@@ -151,10 +151,12 @@ case object DebugModuleHartSelKey extends Field(DebugModuleHartSelFuncs())
 /* structure for passing hartsel between the "Outer" and "Inner"
  */
 
-class DebugInternalBundle ()(implicit val p: Parameters) extends ParameterizedBundle()(p) {
+class DebugInternalBundle (val nComponents: Int)(implicit val p: Parameters) extends ParameterizedBundle()(p) {
   val resumereq    = Bool()
   val hartsel      = UInt(10.W)
   val ackhavereset = Bool()
+  val hasel        = Bool()
+  val hamask       = Vec(nComponents, Bool())
```

Comment:
Hmm... this is all going across a not-so-clean asynchronous crossing. Perhaps it makes sense to shadow the HASEL and HAMASK registers in the proper domain vs doing it this way? Or, to just declare that these will be static so that they don't need to be synchronized?

Reply:
Both of these new items are needed in both domains.  They can't be static because a change to hamask will affect the read values of fields in dmstatus.  hartsel, hasel, and hamask all need to be transmitted from Outer to Inner whenever any one of them changes.

Reply:
yeah, makes sense. Just wanted to be sure you understood how this bundle was used